src/category_theory/limits/shapes/multiequalizer.lean

```diff
@@ -0,0 +1,452 @@
+
+open category_theory
+
+universes v u
+
+/-- The type underlying the multiequalizer diagram. -/
+@[nolint unused_arguments]
+inductive walking_multicospan {α β : Type v} (fst snd : β → α) : Type v
+| left : α → walking_multicospan
+| right : β → walking_multicospan
+
+/-- The type underlying the multiecoqualizer diagram. -/
+@[nolint unused_arguments]
+inductive walking_multispan {α β : Type v} (fst snd : α → β) : Type v
+| left : α → walking_multispan
+| right : β → walking_multispan
+
+namespace walking_multicospan
+
+variables {α β : Type v} {fst snd : β → α}
+
+instance [inhabited α] : inhabited (walking_multicospan fst snd) :=
+⟨left (default _)⟩
+
+/-- Morphisms for `walking_multicospan`. -/
+inductive hom : Π (a b : walking_multicospan fst snd), Type v
+| id (A)  : hom A A
+| fst (b) : hom (left (fst b)) (right b)
+| snd (b) : hom (left (snd b)) (right b)
+
+instance {a : walking_multicospan fst snd} : inhabited (hom a a) :=
+⟨hom.id _⟩
+
+/-- Composition of morphisms for `walking_multicospan`. -/
+def hom.comp : Π {A B C : walking_multicospan fst snd} (f : hom A B) (g : hom B C),
+  hom A C
+| _ _ _ (hom.id X) f := f
+| _ _ _ (hom.fst b) (hom.id X) := hom.fst b
+| _ _ _ (hom.snd b) (hom.id X) := hom.snd b
+
+instance : small_category (walking_multicospan fst snd) :=
+{ hom := hom,
+  id := hom.id,
+  comp := λ X Y Z, hom.comp,
+  id_comp' := begin
+    rintro (_|_) (_|_) (_|_|_),
+    tidy
+  end,
+  comp_id' := begin
+    rintro (_|_) (_|_) (_|_|_),
+    tidy
+  end,
+  assoc' := begin
+    rintro (_|_) (_|_) (_|_) (_|_) (_|_|_) (_|_|_) (_|_|_),
+    tidy,
+  end }
+
+end walking_multicospan
+
+namespace walking_multispan
+
+variables {α β : Type v} {fst snd : α → β}
+
+instance [inhabited α] : inhabited (walking_multispan fst snd) :=
+⟨left (default _)⟩
+
+/-- Morphisms for `walking_multispan`. -/
+inductive hom : Π (a b : walking_multispan fst snd), Type v
+| id (A)  : hom A A
+| fst (a) : hom (left a) (right (fst a))
+| snd (a) : hom (left a) (right (snd a))
+
+instance {a : walking_multispan fst snd} : inhabited (hom a a) :=
+⟨hom.id _⟩
+
+/-- Composition of morphisms for `walking_multispan`. -/
+def hom.comp : Π {A B C : walking_multispan fst snd} (f : hom A B) (g : hom B C),
+  hom A C
+| _ _ _ (hom.id X) f := f
+| _ _ _ (hom.fst a) (hom.id X) := hom.fst a
+| _ _ _ (hom.snd a) (hom.id X) := hom.snd a
+
+instance : small_category (walking_multispan fst snd) :=
+{ hom := hom,
+  id := hom.id,
+  comp := λ X Y Z, hom.comp,
+  id_comp' := begin
+    rintro (_|_) (_|_) (_|_|_),
+    tidy
+  end,
+  comp_id' := begin
+    rintro (_|_) (_|_) (_|_|_),
+    tidy
+  end,
+  assoc' := begin
+    rintro (_|_) (_|_) (_|_) (_|_) (_|_|_) (_|_|_) (_|_|_),
+    tidy,
+  end }
+
+end walking_multispan
+
+/-- This is a structure encapsulating the data necessary to define a `multicospan`. -/
+@[nolint has_inhabited_instance]
+structure multicospan_index (C : Type u) [category.{v} C] :=
+(α β : Type v)
+(f s : β → α)
```

Comment:
These names are not very mnemonic. Also, should `α` and `β` be bundled?

Reply:
I think I'll switch them to `L` resp. `R` (for left resp. right). Is there a particular reason you think they should not be bundled?

In the (hopefully near?) future where we have pseudofunctors, I can imagine wanting to talk about the category of `multicospan_index`s and to a morphism in this category, obtain a functor between `walking_multicospan`s which fit together into a pseudofunctor. This might be an argument for bundling.